WalletWasabi/Blockchain/BlockFilters/IndexBuilderService.cs

```diff
@@ -320,46 +308,37 @@ private void BlockNotifier_OnBlock(object? sender, Block e)
 
 	public (Height bestHeight, IEnumerable<FilterModel> filters) GetFilterLinesExcluding(uint256 bestKnownBlockHash, int count, out bool found)
 	{
-		using (IndexLock.Lock())
+		found = false; // Only build the filter list from when the known hash is found.
+		var filters = new List<FilterModel>();
+		foreach (var filter in Index)
 		{
-			found = false; // Only build the filter list from when the known hash is found.
-			var filters = new List<FilterModel>();
-			foreach (var filter in Index)
+			if (found)
 			{
-				if (found)
-				{
-					filters.Add(filter);
-					if (filters.Count >= count)
-					{
-						break;
-					}
-				}
-				else
+				filters.Add(filter);
+				if (filters.Count >= count)
 				{
-					if (filter.Header.BlockHash == bestKnownBlockHash)
-					{
-						found = true;
-					}
+					break;
 				}
 			}
-
-			if (Index.Count == 0)
-			{
-				return (Height.Unknown, Enumerable.Empty<FilterModel>());
-			}
 			else
 			{
-				return ((int)Index[^1].Header.Height, filters);
+				if (filter.Header.BlockHash == bestKnownBlockHash)
+				{
+					found = true;
+				}
 			}
 		}
+
+		if (Index.Count == 0)
+		{
+			return (Height.Unknown, Enumerable.Empty<FilterModel>());
+		}
+		return ((int)Index.Last().Header.Height, filters);
```

Comment:
Yes, here `Index.Last()` can be newer and that's perfectly okay because we what to return the **current** best height and not what was the best height when we created the filters list.